osfoffline/application/background.py

```diff
@@ -114,32 +93,47 @@ def start_observing_osf_folder(self):
 
         try:
             self.observer.start()  # start
-        except OSError:
+        except OSError as e:
             # FIXME: Document these limits and provide better user notification.
             #    See http://pythonhosted.org/watchdog/installation.html for limits.
-            logging.warning('limit of watched items reached')
+            raise RuntimeError('Limit of watched items reached') from e
```

Comment:
Hadn't know python3 had `raise from`. Always fun to learn a new trick, however small. :)